corehq/apps/es/apps.py

```diff
@@ -32,9 +36,29 @@ class ElasticApp(ElasticDocumentAdapter):
     def mapping(self):
         return get_adapter_mapping(self)
 
-    @classmethod
-    def from_python(cls, doc):
-        return from_dict_with_possible_id(doc)
+    def from_python(self, app):
+        """
+        Takes in an ``Application`` object or an app dict
+        and applies required transformation to make it suitable for ES.
+        The function is replica of ``transform_app_for_es`` with added support for user objects.
+        In future all references to  ``transform_app_for_es`` will be replaced by `from_python`
+
+        :param app: an instance of ``Application`` or ``dict`` which is ``Application.to_json()``
+
+        :raises TypeError: if object passes in not instance of ``ApplicationBase``
+        """
+        from corehq.apps.app_manager.models import ApplicationBase
+        from corehq.apps.app_manager.util import get_correct_app_class
+
+        if isinstance(app, dict):
+            app_obj = get_correct_app_class(app).wrap(app)
+        elif isinstance(app, ApplicationBase):
+            app_obj = app
+        else:
+            raise TypeError(f"Unknown type {type(app)}")
+        app_obj['@indexed_on'] = json_format_datetime(datetime.utcnow())
```

Comment:
seems odd to do this on the object and not on the dict version of it

Reply:
https://github.com/dimagi/commcare-hq/pull/32660/commits/11e06b400c99fc7d1f93a42a1cfd21f0e73d07c5